resources/kcp/charts/kyma-environment-broker/templates/environments-cleanup-job.yaml

```diff
@@ -1,57 +1,62 @@
+        cronjob: {{$job.cronJobName}}
     spec:
       template:
         metadata:
-          name: kyma-environments-cleanup
+          name: {{$job.imageName}}
           labels:
-            cronjob: environments-cleanup
-          {{- if eq .Values.global.database.embedded.enabled false }}
+            cronjob: {{$job.cronJobName}}
+          {{- if eq $.Values.global.database.embedded.enabled false }}
           annotations:
             sidecar.istio.io/inject: "false"
           {{ end }}
         spec:
-          serviceAccountName: kcp-kyma-environment-broker
+          shareProcessNamespace: true
+          {{- if $job.account }}
+          serviceAccountName: {{$job.account}}
+          {{- end }}
+          {{- with $.Values.deployment.securityContext }}
           securityContext:
-            runAsUser: 2000
+            {{ toYaml . }}
+          {{- end }}
           restartPolicy: Never
-          shareProcessNamespace: true
           containers:
-          - name: environments-cleanup
-            image: "{{ .Values.global.images.containerRegistry.path }}/{{ .Values.global.images.kyma_environments_cleanup_job.dir }}kyma-environments-cleanup-job:{{ .Values.global.images.kyma_environments_cleanup_job.version }}"
+          - name: {{ $job.cronJobName }}
+            image: "{{ $.Values.global.images.containerRegistry.path }}/{{ $job.dir }}{{ $job.imageName }}:{{ $job.version }}"
             imagePullPolicy: IfNotPresent
             env:
-              {{if eq .Values.global.database.embedded.enabled false}}
+              {{- range $key, $val := $job.envs }}
+              - name: {{ $key }}
+                value: {{ $val }}
+              {{- end}}
+              - name: APP_PROVISIONER_URL
+                value: "{{ $.Values.provisioner.URL }}"
+              {{if eq $.Values.global.database.embedded.enabled false}}
               - name: DATABASE_EMBEDDED
                 value: "false"
               {{end}}
-              - name: APP_MAX_AGE_HOURS
-                value: "{{ .Values.environmentsCleanup.maxAge }}"
-              - name: APP_LABEL_SELECTOR
-                value: "{{ .Values.environmentsCleanup.labelSelector }}"
-              - name: APP_GARDENER_PROJECT
-                value: "{{ .Values.gardener.project }}"
-              - name: APP_GARDENER_KUBECONFIG_PATH
-                value: "{{.Values.gardener.kubeconfigPath}}"
-              - name: APP_PROVISIONER_URL
-                value: "{{ .Values.provisioner.URL }}"
+              {{if eq $.Values.global.database.embedded.enabled true}}
+              - name: DATABASE_EMBEDDED
+                value: "true"
+              {{end}}
               - name: APP_DATABASE_SECRET_KEY
                 valueFrom:
                   secretKeyRef:
-                    name: "{{ .Values.global.database.managedGCP.encryptionSecretName }}"
+                    name: "{{ $.Values.global.database.managedGCP.encryptionSecretName }}"
```

Comment:
What is difference?

Reply:
Dollar sign. It is needed when context changes in Helm.

Reply:
But it wasn't used so far and worked well, why do we change it?

Reply:
Either first or second line of the file changes which variables and how they need to be accessed.